hibernate-core/src/test/java/org/hibernate/orm/test/mapping/type/java/DateSubclassesUnwrapTest.java

```diff
@@ -0,0 +1,252 @@
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.orm.test.mapping.type.java;
+
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.time.ZonedDateTime;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.TimeZone;
+
+import org.hibernate.HibernateException;
+import org.hibernate.dialect.Dialect;
+import org.hibernate.dialect.H2Dialect;
+import org.hibernate.engine.jdbc.LobCreator;
+import org.hibernate.engine.jdbc.NonContextualLobCreator;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.type.descriptor.WrapperOptions;
+import org.hibernate.type.descriptor.java.CalendarDateJavaType;
+import org.hibernate.type.descriptor.java.CalendarJavaType;
+import org.hibernate.type.descriptor.java.CalendarTimeJavaType;
+import org.hibernate.type.descriptor.java.DateJavaType;
+import org.hibernate.type.descriptor.java.InstantJavaType;
+import org.hibernate.type.descriptor.java.JdbcDateJavaType;
+import org.hibernate.type.descriptor.java.JdbcTimeJavaType;
+import org.hibernate.type.descriptor.java.JdbcTimestampJavaType;
+import org.hibernate.type.descriptor.java.LocalDateJavaType;
+import org.hibernate.type.descriptor.java.LocalDateTimeJavaType;
+import org.hibernate.type.descriptor.java.LocalTimeJavaType;
+import org.hibernate.type.descriptor.java.OffsetDateTimeJavaType;
+import org.hibernate.type.descriptor.java.OffsetTimeJavaType;
+import org.hibernate.type.descriptor.java.ZonedDateTimeJavaType;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class DateSubclassesUnwrapTest {
+
+	private static final WrapperOptions wrapperOptions = new WrapperOptions() {
+		@Override
+		public SharedSessionContractImplementor getSession() {
+			return null;
+		}
+
+		@Override
+		public SessionFactoryImplementor getSessionFactory() {
+			return null;
+		}
+
+		public boolean useStreamForLobBinding() {
+			return false;
+		}
+
+		@Override
+		public int getPreferredSqlTypeCodeForBoolean() {
+			return 0;
+		}
+
+		public LobCreator getLobCreator() {
+			return NonContextualLobCreator.INSTANCE;
+		}
+
+		@Override
+		public TimeZone getJdbcTimeZone() {
+			return null;
+		}
+
+		private final Dialect dialect = new H2Dialect() {
+			@Override
+			public boolean useConnectionToCreateLob() {
+				return false;
+			}
+		};
+
+		@Override
+		public Dialect getDialect() {
+			return dialect;
+		}
+	};
+
+	@Test
+	void testJdbcTimestampJavaType() {
```

Comment:
```suggestion
@SessionFactory
public class DateSubclassesUnwrapTest {
	@Test
	void testJdbcTimestampJavaType(SessionFactoryScope scope) {
		final WrapperOptions wrapperOptions = scope.getSessionFactory().getWrapperOptions();
```

You can avoid creating this anonymous `WrapperOptions` by using the one in session factory.